_posts/2016-03-16-AngularTS.md

```diff
@@ -0,0 +1,413 @@
+To install the angular tsd files we use the tsd manager.
+To use the tsd manager we install it with:
+{% highlight bash %}
+npm install tsd -g
+{% endhighlight %}
+
+Afterwards install angular with:
+{% highlight bash %}
+tsd install angular --resolve --save 
+'--resolve' will include all dependencies angular needs.
+'--save' will include the angular tsd into the tsd.json file.
+{% endhighlight %}
+
+When installed you have a typings folder where you can find all your tsd files. 
+In the tsd.d.ts file you will see the references the application will use for angular. 
+It is possible that you have to declare the referencing on the top of your file.
+{% highlight bash %}
+/// <reference path="../../typings/angularjs/angular.d.ts" />
+/// <reference path="../../typings/angularjs/angular-route.d.ts" />
+{% endhighlight %}
+
+## Modules
+
+### Angular Modules
+Modules are here to help us modularize our code.
+But it is important to know that if you’re not planning to share or declare related functionality.
+It is best practice to use only one main module as the root of your application. 
+To let the module know the existence of every component, they have to register themselves.
+Below every component declaration you will see a registration to the module. 
+When registering the module you have to add all the libraries you want to depend upon.
+In this example we inject the routing service for navigation. 
+
+{% highlight javascript %}
+module JWorks {
+    "use strict";
+
+    angular
+       .module("jworks360", ['ngRoute'])
+     }
+{% endhighlight %}
+
+### Internal TypeScript Modules
+These modules are similar to namespaces.
+You can define an unique namespace around your code.
+This will encapsulate variables, interfaces, and classes. 
+TypeScript supports sub namespaces for further encapsulation.
+
+{% highlight javascript %}
+module JWorks {
+    "use strict";
+}
+{% endhighlight %}
+
+#### Transpiled JavaScript
+
+{% highlight javascript %}
+var JWorks;
+(function(JWorks){
+
+})(JWorks || (JWorks = {}));
+{% endhighlight %}
+
+To encapsulate our code, the module will transpile to an IIFE around our components.
+This will avoid global code which helps prevent variables and function declarations from living longer than expected in the global scope, which also helps avoid variable collisions.
+
+
+## Entity Class
+Now that TypeScript supports OO, we can analyse our business problem and define the business objects into entity classes.
+When you analyse and define these entities you can define which properties and methods each entity needs.
+If you have a couple entities, you can even establish a relationship.
+This will provide a clear view on what you want to achieve and have the possibility to create multiple instances of these classes. 
+When building an entity class you can optionally define an interface to show what’s the intent. 
+
+{% highlight javascript %}
```

Comment:
Isn't there a TypeScript highlight available? Looks a bit odd with the colons marked in red.